head/src/main/kotlin/io/qalipsis/core/head/model/converter/CampaignConfigurationConverterImpl.kt

```diff
@@ -49,7 +51,24 @@ internal class CampaignConfigurationConverterImpl(
     protected fun generateCampaignKey(tenant: String): String = idGenerator.long()
 
     private fun convertScenarioRequestsToConfigurations(scenarios: Map<ScenarioName, ScenarioRequest>): Map<ScenarioName, ScenarioConfiguration> {
-        return scenarios.mapValues { ScenarioConfiguration(it.value.minionsCount, defineExecutionProfileConfiguration(it.value)) }
+        return scenarios.mapValues {
+            ScenarioConfiguration(
+                minionsCount = it.value.minionsCount,
+                executionProfileConfiguration = defineExecutionProfileConfiguration(it.value),
+                zones = it.value.zones?.also(this::checkAndUpdateZone) ?: emptyMap()
+            )
+        }
+    }
+
+    private fun checkAndUpdateZone(zones: Map<String, Int>){
+        //We check if all the keys provide for zones are supported
+        require(headConfiguration.zones.map { it.key }.containsAll(zones.keys)){
+            "All the keys used in the zones argument of the scenario configuration must be define in your configuration"
```

Comment:
Please list the unknown zones in the message:
"The requested zones $unknownZones do not exist"